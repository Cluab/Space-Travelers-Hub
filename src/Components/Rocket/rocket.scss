.rocket {
  font-family: Roboto, sans-serif;
  display: flex;
  gap: 1rem;

  & .rocket-img {
    width: 350px;
  }

  & > .rocket-details {
    display: flex;
    flex-direction: column;
    gap: 1rem;

    & > h2 {
      font-weight: 500;
      font-size: 1.4em;
    }

    & > p {
      font-weight: 400;
      font-size: 1em;
      line-height: 24px;
      letter-spacing: 0.7px;
    }
  }

  @media only screen and (max-width: 768px) {
    flex-direction: column;
  }
}

.reserved-tag {
  background-color: #16a1b7;
  color: rgb(241 241 241);
  padding: 0.25rem 0.3rem;
  outline: inherit;
  border-radius: 5px;
  font-weight: 500;
  font-size: 12px;
  line-height: 24px;
  margin-right: 0.5rem;
}

.reserve {
  background-color: #007bff;
  color: rgb(241 241 241);
  border: none;
  padding: 0.7rem;
  cursor: pointer;
  outline: inherit;
  width: 150px;
  border-radius: 4px;
  font-weight: 400;
  font-size: 18px;
  font-family: Roboto, sans-serif;
}

.reserved {
  background-color: #fff;
  color: #808080;
  border: 1px solid #808080;
  padding: 0.7rem;
  cursor: pointer;
  outline: inherit;
  width: 180px;
  border-radius: 4px;
  font-weight: 400;
  font-size: 18px;
  font-family: Roboto, sans-serif;
}
