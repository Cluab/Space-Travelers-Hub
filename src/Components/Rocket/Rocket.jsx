import './rocket.scss';
import { PropTypes } from 'prop-types';

const Rocket = ({
  rocketImage,
  rocketName,
  description,
  reserved,
}) => (
  <article className="rocket">
    <figure>
      <img className="rocket-img" src={rocketImage} alt="rocket" />
    </figure>
    <div className="rocket-details">
      <h2>{rocketName}</h2>
      <p>
        { reserved && <span className="reserved-tag">Reserved</span> }
        {description}
      </p>
      {
        reserved
          ? <button type="button" className="reserve">Reserve Rocket</button>
          : <button type="button" className="reserved">Cancel Reservation</button>
      }
    </div>
  </article>
);

Rocket.propTypes = {
  rocketImage: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
