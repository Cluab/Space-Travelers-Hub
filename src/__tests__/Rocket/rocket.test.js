import { act, render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import Rocket from '../../Components/Rocket/Rocket';
import rocketsReducer, { handleReservation } from '../../Redux/Rockets/rockets';

describe('Rocket component', () => {
  const store = configureStore({ reducer: { rockets: rocketsReducer } });

  const data = [
    {
      id: 1,
      rocketName: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
      rocketImage: 'https://imgur.com/DaCfMsj.jpg',
      reserved: false,
    },
    {
      id: 2,
      rocketName: 'Falcon 9',
      description: 'Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.',
      rocketImage: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
      reserved: false,
    },
  ];

  it('should match the snapshot', () => {
    const rocketComponent = TestRenderer.create(
      <Provider store={store}>
        <Rocket
          rocket={data[0]}
          handleReservation={() => store.dispatch(handleReservation(data[0].id))}
        />
      </Provider>,
    ).toJSON();

    expect(rocketComponent).toMatchSnapshot();
  });

  it('should display Rocket details', async () => {
    const { container } = await act(() => render(
      <Provider store={store}>
        <Rocket
          rocket={data[1]}
          handleReservation={() => store.dispatch(handleReservation(data[1].id))}
        />
      </Provider>,
    ));

    const rocketDetailsEl = container.querySelector('.rocket-details');
    const title = rocketDetailsEl.querySelector('h2').textContent;
    const description = rocketDetailsEl.querySelector('p').textContent;

    expect(title).toBe(data[1].rocketName);
    expect(description).toBe(data[1].description);
  });
});
