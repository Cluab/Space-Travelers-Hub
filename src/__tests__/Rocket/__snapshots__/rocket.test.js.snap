// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component should render rocket component 1`] = `
<article
  className="rocket"
>
  <figure>
    <img
      alt="rocket"
      className="rocket-img"
      src="https://imgur.com/DaCfMsj.jpg"
    />
  </figure>
  <div
    className="rocket-details"
  >
    <h2>
      Falcon 1
    </h2>
    <p>
      The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
    </p>
    <button
      className="reserve"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</article>
`;
